lookout/style/typos/analyzer.py

```diff
@@ -284,20 +336,37 @@ def reconstruct_identifier(tokenizer: TokenParser, pred_tokens: List[str], ident
         return "".join(res)
 
     @classmethod
+    @with_uasts_and_contents(unicode=True)
     def train(cls, ptr: ReferencePointer, config: Mapping[str, Any], data_service: DataService,
-              **data) -> AnalyzerModel:
+              files: Iterator[File], **data) -> AnalyzerModel:
         """
         Generate a new model on top of the specified source code.
 
         :param ptr: Git repository state pointer.
         :param config: Configuration of the training of unspecified structure.
         :param data_service: The channel to the data service in Lookout server to query for \
                              UASTs, file contents, etc.
+        :param files: iterator of File records from the data service.
         :param data: Extra data passed into the method. Used by the decorators to simplify \
                      the data retrieval.
         :return: Instance of `AnalyzerModel` (`model_type`, to be precise).
         """
-        return DummyAnalyzerModel()
+        _log = logging.getLogger(cls.__name__)
+        train_config = cls._load_config(config)
+        _log.info("train %s %s %s %s", __version__, ptr.url, ptr.commit,
+                  pformat(train_config, width=4096, compact=True))
+        model = IdTyposModel()
+        if not train_config["store_identifiers"]:
+            return model
+        for _, files in files_by_language(files).items():
+            for file in filter_files(
+                    files=files, line_length_limit=train_config["line_length_limit"],
+                    overall_size_limit=train_config["overall_size_limit"], log=_log):
+                model.update({
+                    node.token for node in uast2sequence(file.uast)
```

Comment:
I think that it is possible to use `self._get_identifiers()` here because they both do the same thing.
```suggestion
                    node.token for node in self._get_identifiers(file.uast, [])
```

Is it correct?

Reply:
Yes, it should be